component/api/src/main/java/org/exoplatform/social/metadata/MetadataService.java

```diff
@@ -193,6 +193,12 @@ MetadataItem createMetadataItem(MetadataObject metadataObject,
    */
   List<MetadataType> getMetadataTypes();
 
+  /**
+   * @param type metadata type {@link Metadata} type
+   * @return {@link List} of Managed {@link Metadata}
+   */
+  List<Metadata> getMetadatasByType(long type);
```

Comment:
we should use name as parameter
we should use offset/limit and use limit=100 to get the whole list